tasks/db_deploy/migrations/migrate_versions_4_to_5/migrate_sql.py

```diff
@@ -207,32 +204,26 @@ def reconcile_s3_object_partition_sql(partition_name: str) -> TextClause:
     Returns:
         (sqlalchemy.sql.element.TextClause): SQL for creating reconcile_s3_object partition table.
     """
-    try:
-        if not re.match("^[\w+]+$", partition_name):  # noqa: W605
-            raise ValueError(f"Table name {partition_name} is invalid.")
-    except TypeError:
-        raise ValueError("Table name must be a string and cannot be None.")
-
```

Comment:
why are we removing this check?

Reply:
It's being moved to the calling function to remove business logic from the sql functions.

Reply:
It is already covered by shared_reconciliation.get_partition_name_from_bucket_name, which forces the string to be alpha-numeric, plus underscores. I could be convinced to put this back in, but it would need to be pulled out of this function, and would need to be better documented.